website/database.py

```diff
@@ -234,8 +234,10 @@ def filtered_programs_for_user(self, username, level=None, adventure=None, submi
 
         # FIXME: Query by index, the current behavior is slow for many programs
         # (See https://github.com/hedyorg/hedy/issues/4121)
-        programs = dynamo.GetManyIterator(PROGRAMS, {"username": username},
-                                          reverse=True, batch_size=limit, pagination_token=pagination_token)
+        programs = PROGRAMS.get_many({"username": username},
+                                     reverse=True, limit=limit, pagination_token=pagination_token)
+        # programs = dynamo.GetManyIterator(PROGRAMS, {"username": username},
+        #                                   reverse=True, batch_size=limit, pagination_token=pagination_token)
```

Comment:
This doesn't do the same as it did before.

This will retrieve `limit` items, then filter them client-side (leaving you potentially with `<limit` items).

On the other hand, the `GetManyIterator` will keep on pulling items from the database until you stop pulling (which we will do only once we have exactly `limit` items, or have reached the end of the list).